integration/src/main.rs

```diff
@@ -0,0 +1,57 @@
+// TODO: Things to figure out:
+//  - Tests all have to be called manually, that sucks!
+//  - Tests run after each other, one failure prevents next one from running
+//  - No status output about success
```

Comment:
I've never needed a custom test harness so far, but I assume someone might have developed one at some point? If that was me, test concurrency would be one of my first concerns, so maybe we can look for something in the wild.

Reply:
I've definitely seen at least one advertised on /r/rust recently. But I think we should be able to go without.

Reply:
I've also seen this bit on page 86 of the "Rust for Rustaceans" book:

> A test harness is generated for each file in `tests/`. Test harnesses are not generated for files in subdirectories under `tests/` to allow you to have shared submodules for your tests. NOTE: If you explicitly want a test harness for a file in a subdirectory, you can opt in to that by calling the file `main.rs`.

I don't know if that helps or not for this review, but it did seem like a useful tidbit.